packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -164,82 +272,82 @@ const Switch = ({
     }
   }, [isChecked]);
 
-  const id = useUniqueId(_id);
+  const inputId = useUniqueId(_id);
 
-  const hintId = hint ? `${id}-hint` : undefined;
-  const labelElementId = `${id}-label`;
+  const hintId = hint ? `${inputId}-hint` : undefined;
+  const labelElementId = `${inputId}-label`;
 
-  const handleOnChange = useCallback(
-    (_: SyntheticEvent<Element, Event>, checked: boolean) => {
-      setInternalSwitchChecked(checked);
+  const handleOnChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
+    (event) => {
+      const target = event?.target;
+      const { checked, value } = target;
+      setInternalSwitchChecked(!internalSwitchChecked);
```

Comment:
When setting state in React, use the function in a state setter rather than the value. There will be timing issues otherwise, and it will cause unnecessary re-renders:
```suggestion
  const handleOnChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
    (event) => {
      const target = event?.target;
      const { checked, value } = target;
      setInternalSwitchChecked(internalSwitchChecked => !internalSwitchChecked);
```

Reply:
Don't think I even need this. Think I can just set it to `checked` here